pkg/cloud/aws/actuators/machine/actuator_test.go

```diff
@@ -412,3 +412,132 @@ func TestBuildEC2Filters(t *testing.T) {
 		t.Errorf("failed to buildEC2Filters. Expected: %+v, got: %+v", expected, got)
 	}
 }
+
+func TestAvailabiltyZone(t *testing.T) {
+	cases := []struct {
+		name             string
+		availabilityZone string
+		subnet           string
+	}{
+		{
+			name:             "availability zone only",
+			availabilityZone: "us-east-1a",
+		},
+		{
+			name:   "subnet only",
+			subnet: "subnet-b46032ec",
+		},
+		{
+			name:             "availability zone and subnet",
+			availabilityZone: "us-east-1a",
+			subnet:           "subnet-b46032ec",
+		},
+	}
+
+	codec, err := providerconfigv1.NewCodec()
+	if err != nil {
+		t.Fatalf("unable to build codec: %v", err)
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
what is this actually validating, I couldn't find any assertion?

Reply:
It asserts the placement trough the mock assertion, line 487.